src/utils/cache.h

```diff
@@ -84,8 +87,8 @@ class LruCache {
 
   // Looks up and pins the element by key. Returns nullptr if not found.
   // If found, brings the element to the head of the queue (makes it last to
-  // evict).
-  V* Lookup(K key) {
+  // evict); furthermore, a call to Unpin must be made for each such element.
```

Comment:
That cache pinning is pretty standard thing, so it sounds almost like "malloc(), allocates memory; futhermore, a call to free() must be made for those elements".

I think what would be useful is usage example in comment on the top of the file, and for LookupAndPin() and Unpin() functions comments that `LruCacheLock` is much more recommended.

Reply:
I'd never heard of cache pinning before, it's certainly not a thing in Python land. Maybe it's only a thing in compiled/optimized languages?

And yes "`malloc` allocates and requires `free`" is absolutely crucial documentation to malloc. Even though "[everyone knows](https://xkcd.com/1053/)" it, we must never assume such knowledge and always document it. Can you imagine glibc maintainers accepting a change to malloc documentation which removed the explanation that a call to `free` is always required?

With sufficiently good documentation, I think an example is unnecessary, and I think that adding more references to `LruCacheLock` as you suggest is sufficient to avoid needing an example.